docker/docker.go

```diff
@@ -25,6 +26,14 @@ const (
 )
 
 func main() {
+
+	for j := 1; j < len(os.Args); j++ {
+		if os.Args[j] == "-d" || os.Args[j] == "--daemon" {
+			log.Printf("%s is deprecated. Please use daemon command", os.Args[j])
+			os.Args[j] = "daemon"
```

Comment:
Actually, realisation that this suggestion isn't sufficient - see https://github.com/docker/docker/pull/10535#issuecomment-72765137 below.

i.e. you make it impossible to pass global-level arguments if you shuffle the daemon command to the front.

If #10453 was merged, then this PR was rebased on top of it, we'd have a decent way of filtering os.Args into two lists: global args and daemon args. Without that we're still treating api-enable-cors as a global arg when this PR gets merged, which is wrong.